packages/application-shell-connectors/src/components/project-extension-image-regex/project-extension-image-regex.tsx

```diff
@@ -54,24 +73,36 @@ const ProjectExtensionProviderForImageRegex = (props: ProviderProps) => {
 ProjectExtensionProviderForImageRegex.displayName =
   'ProjectExtensionProviderForImageRegex';
 
-const GetProjectExtensionImageRegex = (props: ConsumerProps) => (
-  <Context.Consumer>
-    {(imageRegexContext) => props.render(imageRegexContext)}
-  </Context.Consumer>
-);
+const GetProjectExtensionImageRegex = (props: ConsumerProps) => {
+  //TODO: add a descriptive possible deprecation warning for moving from HOC to Hooks in the future
+  useWarning(
+    false,
+    `@commercetools-frontend/application-shell-connectors: ...`
+  );
+
+  return (
+    <Context.Consumer>
+      {(imageRegexContext) => props.render(imageRegexContext)}
+    </Context.Consumer>
+  );
+};
 GetProjectExtensionImageRegex.displayName = 'GetProjectExtensionImageRegex';
 
 function withProjectExtensionImageRegex<Props extends {}>(
-  propKey = 'imageRegexData'
+  propKey: 'imageRegexData'
```

Comment:
I guess we need to keep the equal sign so the prop keeps being optional.
Otherwise, it would be required and the only value TS would accept would be `imageRegexData`.

```suggestion
  propKey = 'imageRegexData'
```